core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -418,9 +420,13 @@ trait DerivationSchema[R] {
 
 object GenericSchema {
 
-  def effectfulExecutionError(path: List[Either[String, Int]], e: Throwable): ExecutionError =
+  def effectfulExecutionError(
```

Comment:
Not related to your code, but this one seems only used in `Executor`, how about moving it there?